codebuild/builder.py

```diff
@@ -490,24 +497,150 @@ def run_build(build_spec, is_dryrun):
     source_dir = os.environ.get("CODEBUILD_SRC_DIR", os.getcwd())
     sources = [os.path.join(source_dir, file) for file in glob.glob('**/*.c')]
 
-    def _flatten_command(command):
+    built_projects = []
+
+    def _flatten_command(*command):
         # Process out lists
         new_command = []
-        for e in command:
-            e_type = type(e)
+        def _proc_segment(command_segment):
+            e_type = type(command_segment)
             if e_type == str:
-                new_command.append(e)
-            elif e_type == list:
-                new_command.extend(e)
+                new_command.append(command_segment)
+            elif e_type == list or e_type == tuple:
+                for segment in command_segment:
+                    _proc_segment(segment)
+        _proc_segment(command)
         return new_command
 
-    def _log_command(command):
-        print('>', ' '.join(_flatten_command(command)), flush=True)
+    def _log_command(*command):
+        print('>', ' '.join(_flatten_command(*command)), flush=True)
 
-    def _run_command(command):
-        _log_command(command)
+    def _run_command(*command):
+        _log_command(*command)
         if not is_dryrun:
-            subprocess.check_call(_flatten_command(command), stdout=sys.stdout, stderr=sys.stderr)
+            subprocess.check_call(_flatten_command(*command), stdout=sys.stdout, stderr=sys.stderr)
+
+    # Helper to run makedirs regardless of dry run status
+    def _mkdir(directory):
+        if not is_dryrun:
+            os.makedirs(directory, exist_ok=True)
+        _log_command("mkdir", "-p", directory)
```

Comment:
Can we log before attempting the makedirs, so if an exception is thrown, we have some info?

Reply:
Sure